app/src/main/java/com/example/habitsmasher/HabitEventDialog.java

```diff
@@ -87,25 +125,54 @@ protected void setCancelButtonListener() {
     protected abstract void setConfirmButtonListener();
 
     /**
-     * Opens the calendar dialog used for date selection
+     * Sets up the listener for the location button
      */
-    protected void openDatePickerDialog() {
-        DatePickerDialogFragment datePickerDialogFragment = new DatePickerDialogFragment(new DatePickerDialog.OnDateSetListener() {
-            /**
-             * Sets the text of the date select view to reflect selected date
-             * @param view
-             * @param year year of selected date
-             * @param month month of selected date (integer from 0 to 11)
-             * @param day day of month of selected date
-             */
+    protected void setLocationButtonListener() {
+        _addLocationButton.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onDateSet(DatePicker view, int year, int month, int day) {
-                int correctedMonth = month + 1;
-                String date = day + "/" + correctedMonth + "/" + year;
-                _eventDateText.setText(date);
+            public void onClick(View view) {
+                // checks location permissions
+                int coarsePermission = ContextCompat.checkSelfPermission(getContext(),
+                        Manifest.permission.ACCESS_COARSE_LOCATION);
+                int precisePermission = ContextCompat.checkSelfPermission(getContext(),
+                        Manifest.permission.ACCESS_FINE_LOCATION);
+                // if granted already
+                if (coarsePermission == PackageManager.PERMISSION_GRANTED &&
+                        precisePermission == PackageManager.PERMISSION_GRANTED) {
+                    // do location stuff
+                    handleLocationSelection();
+                }
+                else {
+                    // request permissions
+                    String[] permissionsArray = {Manifest.permission.ACCESS_COARSE_LOCATION,
+                            Manifest.permission.ACCESS_FINE_LOCATION};
+                    _requestPermissionsLauncher.launch(permissionsArray);
+                }
```

Comment:
This snippet essentially checks whether a user has granted location permissions to the app